File: web/src/api/projectmgr/reportNotify.js
import service from '@/utils/request'

export const createReportNotifyChannel = (data) => {
  return service({
    url: '/nt/createReportNotifyChannel',
    method: 'post',
    data
  })
}

export const deleteReportNotifyChannel = (params) => {
  return service({
    url: '/nt/deleteReportNotifyChannel',
    method: 'delete',
    params
  })
}

export const deleteReportNotifyChannelByIds = (params) => {
  return service({
    url: '/nt/deleteReportNotifyChannelByIds',
    method: 'delete',
    params
  })
}

export const updateReportNotifyChannel = (data) => {
  return service({
    url: '/nt/updateReportNotifyChannel',
    method: 'put',
    data
  })
}

export const findReportNotifyChannel = (params) => {
  return service({
    url: '/nt/findReportNotifyChannel',
    method: 'get',
    params
  })
}

export const getReportNotifyChannelList = (params) => {
  return service({
    url: '/nt/getReportNotifyChannelList',
    method: 'get',
    params
  })
}

export const getAutoReportNotifyStatus = (params) => {
  return service({
    url: '/nt/getAutoReportNotifyStatus',
    method: 'get',
    params
  })
}

File: server/router/projectmgr/report_notify.go
package projectmgr

import (
	"github.com/gin-gonic/gin"
	"github.com/test-instructor/yangfan/server/v2/middleware"
)

type ReportNotifyRouter struct{}

func (s *ReportNotifyRouter) InitReportNotifyRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	ntRouter := Router.Group("nt").Use(middleware.OperationRecord())
	ntRouterWithoutRecord := Router.Group("nt")
	{
		ntRouter.POST("createReportNotifyChannel", ntApi.CreateReportNotifyChannel)
		ntRouter.DELETE("deleteReportNotifyChannel", ntApi.DeleteReportNotifyChannel)
		ntRouter.DELETE("deleteReportNotifyChannelByIds", ntApi.DeleteReportNotifyChannelByIds)
		ntRouter.PUT("updateReportNotifyChannel", ntApi.UpdateReportNotifyChannel)
	}
	{
		ntRouterWithoutRecord.GET("findReportNotifyChannel", ntApi.FindReportNotifyChannel)
		ntRouterWithoutRecord.GET("getReportNotifyChannelList", ntApi.GetReportNotifyChannelList)
		ntRouterWithoutRecord.GET("getAutoReportNotifyStatus", ntApi.GetAutoReportNotifyStatus)
	}
}
